### ISSUE TITLE
Need default-libmysqlclient-dev in debian stable

### ISSUE THREAD

**Issue by CicadaSeventeen** [pyenv/pyenv#2833]
When I try to install `Mathics-omnibus[full]` from PyPI, I meet an error that can be solved by installing `default-libmysqlclient-dev`.

Please add that into wiki page so that other can know it.

**Comment by native-api** [comment]
https://github.com/pyenv/pyenv/wiki#suggested-build-environment only lists dependencies for the base Python. Of course, 3rd-party extension packages may require some other packages on the system. The documentation for those packages should be listing those.

We can create a separate Wiki page listing these. Will it be of much use though?

**Comment by CicadaSeventeen** [comment]
Maybe I have not explained clearly. `default-libmysqlclient-dev` for debian (maybe `libmysqlclient-dev` for ubuntu?) is necessary not to build `mathic3` but to install it from `pip/PyPI`. The problem here is that `Mathics-omnibus[full]` relays on `mysqlclient` while it need dev package from host system to build. 

That information could be useful for those have problem to install `mathics` from `pip` for distros that do not have mathics package.

**Comment by native-api** [comment]
I've double-checked, and the MySQL deveopment files are indeed needed to build a 3rd-party module, `mysqlclient`, which is one of `Mathics-omnibus[full]`'s dependencies.

If a precompiled wheel from the current environment is not found at PyPI, Pip builds the package from a source distribution as part of installation (just like Pyenv builds Python from source upon installation). [`mysqlclient` only provides wheels for Windows](https://pypi.org/project/mysqlclient/#files).

```
vmuser@ubuntuvm:~$ pip install -v 'Mathics-omnibus[full]'
<...>
Collecting mysqlclient
  Downloading mysqlclient-2.2.0.tar.gz (89 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.5/89.5 kB 9.5 MB/s eta 0:00:00
  <...>
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  Trying pkg-config --exists mysqlclient
  Command 'pkg-config --exists mysqlclient' returned non-zero exit status 1.
  Trying pkg-config --exists mariadb
  Command 'pkg-config --exists mariadb' returned non-zero exit status 1.
  Traceback (most recent call last):
    File "/home/vmuser/.pyenv/versions/3.10.13/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/home/vmuser/.pyenv/versions/3.10.13/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/vmuser/.pyenv/versions/3.10.13/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-bvsy3rp_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-bvsy3rp_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-bvsy3rp_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 154, in <module>
    File "<string>", line 48, in get_config_posix
    File "<string>", line 27, in find_package_name
  Exception: Can not find valid pkg-config name.
  Specify MYSQLCLIENT_CFLAGS and MYSQLCLIENT_LDFLAGS env vars manually
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/vmuser/.pyenv/versions/3.10.13/bin/python3.10 /home/vmuser/.pyenv/versions/3.10.13/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpwmkwgoha
  cwd: /tmp/pip-install-ix2qqfyh/mysqlclient_ceca2fab496543cea4d3a15ca28c91aa
  Getting requirements to build wheel ... error
error: subprocess-exited-with-error
<...>
```

**Comment by CicadaSeventeen** [comment]
@native-api 

I managed to install `Mathics-omnibus[full]` in a pure `debian` dokcer, so that I have found the packages that needed.

 If I use `pip`, when `--install-recommends`, I need

```
apt install pkg-config default-libmysqlclient-dev python3-pip gcc python3-dev python3-virtualenv
```

Notice that installing by `pip install Mathics-omnibus[full] --break-system-packages` will not work maybe because some system packages of `debian`. `virtualenv` is necessary.

If use `pipx` to install it, the packages needed is
```
apt install pkg-config default-libmysqlclient-dev pipx gcc python3-dev
```
Noting that install `Mathics-omnibus` only make commands like `dmathics` themselves, so manually link commands like `mathics` is necessary.

Besides, since commands like `dmathics ` need `docker` to run, installing `docker.io` or `podman-docker` is needed to run `dmathics` command. Though I instead of running it I prefer to use `mathics` so I do not install `docker` or `podman` here.
